doc/book/methods.md

```diff
@@ -1,62 +1,39 @@
-`hasChildren() : bool`              | Indicates if the current role has children.
-`key() : int`                       | Index of the current role instance.
-`next() : void`                     | Advance to the next role instance.
-`rewind() : void`                   | Seek to the first item in the iterator.
-`valid() : bool`                    | Is the current index valid?
-
-## `Zend\Permissions\Rbac\AbstractRole`
-
-The `AbstractRole` provides the base functionality required by the
-`RoleInterface`, and is the foundation for the `Role` class.
-
-Method signature                               | Description
----------------------------------------------- | -----------
-`addChild(string|RoleInterface $child) : void` | Add a child role to the current instance.
-`addPermission(string $name) : void`           | Add a permission for the current role.
-`getName() : string`                           | Retrieve the name assigned to this role.
-`hasPermission(string $name) : bool`           | Does the role have the given permission?
-`setParent(RoleInterface $parent) : void`      | Assign the provided role as the current role's parent.
-`addParent(RoleInterface $parent) : Role`      | Add a parent role to the current instance.
-`getParent() null|RoleInterface|array`         | Retrieve the current role's parent, or array of parents if more that one exists.
+## `Zend\Permissions\Rbac\Role`
+
+The `Role` provides the base functionality required by the `RoleInterface`.
+
+Method signature                          | Description
+------------------------------------------| -----------
+`__construct(string $name) : void`        | Create a new instance with the provided name.
+`getName() : string`                      | Retrieve the name assigned to this role.
+`addPermission(string $name) : void`      | Add a permission for the current role.
+`hasPermission(string $name) : bool`      | Does the role have the given permission?
+`addChild(RoleInterface $child) : Role`   | Add a child role to the current instance.
+`getChildren() : RoleInterface[]`         | Get all the children roles.
+`addParent(RoleInterface $parent) : Role` | Add a parent role to the current instance.
+`getParents() : RoleInterface[]`          | Get all the parent roles.
 
 ## `Zend\Permissions\Rbac\AssertionInterface`
 
 Custom assertions can be provided to `Rbac::isGranted()` (see below); such
 assertions are provided the `Rbac` instance on invocation.
 
-Method signature            | Description
---------------------------- | -----------
-`assert(Rbac $rbac) : bool` | Given an RBAC, determine if permission is granted.
+Method signature                                                            | Description
+--------------------------------------------------------------------------- | -----------
+`assert(Rbac $rbac, string $permission = null, RoleInterface $role = null)` | Given an RBAC, an optional permission, and optional role determine if permission is granted.
```

Comment:
This somewhat answers my earlier question. My reading of this, though is that:

- The rbac is always passed
- The permission will likely always be passed (as we're asserting a permission)
- Shouldn't the role always be passed? Aren't we asserting the role has the given permission?

Reply:
Read my previous comment.